lib/model.js

```diff
@@ -3913,7 +3921,12 @@ class Model {
     if (this.isNewRecord && createdAtAttr && !this.dataValues[createdAtAttr]) {
       this.dataValues[createdAtAttr] = this.constructor._getDefaultTimestamp(createdAtAttr) || now;
     }
-
+    // Db2 does not allow NULL values for unique columns.
+    // Add dummy values if not provided by test case or user.
+    if (this.sequelize.options.dialect === 'db2' && this.isNewRecord) {
```

Comment:
If possible I'd like to get rid of this somehow

Reply:
To get rid of it, we have to remove all test cases for Db2 server which do not pass a value for unique column or null value is passed for unique column. If two columns will have NULL value, then it can't be unique as it is duplicate value.